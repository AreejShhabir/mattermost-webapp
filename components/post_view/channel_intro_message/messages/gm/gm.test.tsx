// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {archivedChannel, groupChannel, users} from '../test_utils';

import GMIntroMessage from './gm';

const component = (otherProps: any) => {
    return (
        <GMIntroMessage
            currentUserId={'test-user-id'}
            channelProfiles={[]}
            channel={groupChannel}
            {...otherProps}
        />
    );
};

describe('components/post_view/ChannelIntroMessages', () => {
    describe('test Group Channel', () => {
        test('should match snapshot, no profiles', () => {
            expect(shallow(component({}))).toMatchSnapshot();
        });

        test('should match snapshot, with profiles, show boards', () => {
            expect(shallow(component({
                channelProfiles: users,
            }))).toMatchSnapshot();
        });

        test('should match snapshot, archived channel, no boards', () => {
            expect(shallow(component({
                channelProfiles: users,
                channel: archivedChannel,
            }))).toMatchSnapshot();
        });
    });
});
