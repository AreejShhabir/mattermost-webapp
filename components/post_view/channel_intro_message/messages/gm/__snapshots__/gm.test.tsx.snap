// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/post_view/ChannelIntroMessages test Group Channel should match snapshot, no profiles 1`] = `
<p
  className="channel-intro-text"
>
  <MemoizedFormattedMessage
    defaultMessage="This is the start of your group message history with these teammates. Messages and files shared here are not shown to people outside this area."
    id="intro_messages.group_message"
  />
</p>
`;

exports[`components/post_view/ChannelIntroMessages test Group Channel should match snapshot, with profiles, with boards 1`] = `
<Fragment>
  <div
    className="post-profile-img__container channel-intro-img"
  >
    <ProfilePicture
      hasMention={false}
      isEmoji={false}
      isRHS={false}
      key="introprofilepictureuser1"
      popoverPlacement="right"
      size="xl"
      src="/api/v4/users/user1/image?_=0"
      userId="user1"
      wrapperClass=""
    />
    <ProfilePicture
      hasMention={false}
      isEmoji={false}
      isRHS={false}
      key="introprofilepictureguest1"
      popoverPlacement="right"
      size="xl"
      src="/api/v4/users/guest1/image?_=0"
      userId="guest1"
      wrapperClass=""
    />
  </div>
  <p
    className="channel-intro-text"
  >
    <FormattedMarkdownMessage
      defaultMessage="This is the start of your group message history with {names}.\\\\nMessages and files shared here are not shown to people outside this area."
      id="intro_messages.GM"
      values={
        Object {
          "names": "test channel",
        }
      }
    />
  </p>
  <Connect(Component) />
  <Connect(Component)
    show={true}
  />
</Fragment>
`;

exports[`components/post_view/ChannelIntroMessages test Group Channel should match snapshot, with profiles, without boards 1`] = `
<Fragment>
  <div
    className="post-profile-img__container channel-intro-img"
  >
    <ProfilePicture
      hasMention={false}
      isEmoji={false}
      isRHS={false}
      key="introprofilepictureuser1"
      popoverPlacement="right"
      size="xl"
      src="/api/v4/users/user1/image?_=0"
      userId="user1"
      wrapperClass=""
    />
    <ProfilePicture
      hasMention={false}
      isEmoji={false}
      isRHS={false}
      key="introprofilepictureguest1"
      popoverPlacement="right"
      size="xl"
      src="/api/v4/users/guest1/image?_=0"
      userId="guest1"
      wrapperClass=""
    />
  </div>
  <p
    className="channel-intro-text"
  >
    <FormattedMarkdownMessage
      defaultMessage="This is the start of your group message history with {names}.\\\\nMessages and files shared here are not shown to people outside this area."
      id="intro_messages.GM"
      values={
        Object {
          "names": "test channel",
        }
      }
    />
  </p>
  <Connect(Component) />
  <Connect(Component)
    show={true}
  />
</Fragment>
`;
